src/crystal/system/win32/file.cr

```diff
@@ -63,21 +63,20 @@ module Crystal::System::File
   end
 
   private def self.posix_to_open_opts(flags : Int32, perm : ::File::Permissions)
-    access = disposition = attributes = 0
+    disposition = 0
 
-    case flags & (LibC::O_RDONLY | LibC::O_WRONLY | LibC::O_RDWR)
-    when LibC::O_RDONLY then access = LibC::GENERIC_READ
-    when LibC::O_WRONLY then access = LibC::GENERIC_WRITE
-    when LibC::O_RDWR   then access = LibC::GENERIC_READ | LibC::GENERIC_WRITE
-    end
+    access = if flags & LibC::O_WRONLY > 0
+               LibC::GENERIC_WRITE
+             elsif flags & LibC::O_RDWR > 0
+               LibC::GENERIC_READ | LibC::GENERIC_WRITE
+             else
+               LibC::GENERIC_READ
```

Comment:
`O_RDONLY`'s value is actually `0`, so was failing the `> 0` or `!= 0` check, so I just made it the default given that was `0` anyway.

Reply:
Windows only [says](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/open-wopen):

> To specify the file access mode, you must specify either `_O_RDONLY`, `_O_RDWR`, or `_O_WRONLY`. There's no default value for the access mode.

So this check is fine on Windows, but to be precise, these flags are not _bit_ flags at all. POSIX defines also `O_ACCMODE`, to be used like this:

```crystal
access =
  case flags & LibC::O_ACCMODE # unavailable on MSVC
  when LibC::O_RDONLY
  when LibC::O_WRONLY
  when LibC::O_RDWR
  when LibC::O_SEARCH # unavailable on most platforms
  when LibC::O_EXEC   # unavailable on most platforms
  else
    raise "not allowed"
  end
```